### PR TITLE
[webview_flutter] Extract Android implementation into a separate package

### PR DESCRIPTION
**Pull request by mvanbeusekom**
This PR puts the native Android implementation of the platform interface (which was separated in #4302) into its own package, in preparation for moving the main plugin to a federated architecture.

Relevant issue:
 - flutter/flutter#86286

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by BeMacized** [comment]
@flutter/googlebot I consent.

**Comment by BeMacized** [comment]
@googlebot I consent.

File: packages/webview_flutter/webview_flutter_android/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: webview_flutter_android
+description: A Flutter plugin that provides a WebView widget on Android.
+repository: https://github.com/flutter/plugins/tree/master/packages/webview_flutter/webview_flutter_android
+issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+webview%22
+version: 0.0.1
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_android/pubspec.yaml:5]: Same version comment here as for iOS.

File: packages/webview_flutter/webview_flutter_android/example/lib/main.dart [hunk 2]
```diff
@@ -352,3 +364,599 @@ class NavigationControls extends StatelessWidget {
     );
   }
 }
+
+/// A web view widget for showing html content.
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_android/example/lib/main.dart:368]: Same comment here as for iOS about moving to a separate file.

File: packages/webview_flutter/webview_flutter_android/example/integration_test/webview_flutter_test.dart [hunk 3]
```diff
@@ -14,13 +14,13 @@ import 'package:flutter_test/flutter_test.dart';
...
 import 'package:webview_flutter_android/webview_surface_android.dart';
-import 'package:webview_flutter_example/main.dart';
+import 'package:webview_flutter_android_example/main.dart';
 import 'package:webview_flutter_platform_interface/webview_flutter_platform_interface.dart';
 
 void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
 
-  const bool _skipDueIssue86757 = false;
+  const bool _skipDueIssue86757 = true;
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_android/example/integration_test/webview_flutter_test.dart:23]: Nit: _skipDueToIssue86757

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Review by stuartmorgan — APPROVED** [review]
Changes LGTM

**Pull request merged**